pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -184,10 +184,13 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 
 		secret := r.secretForInfinispan(identities, infinispan)
-		err = r.client.Create(context.TODO(), secret)
-		if err != nil {
-			reqLogger.Error(err, "could not find or create identities Secret")
-			return reconcile.Result{}, err
+		if infinispan.Spec.Security.EndpointSecretName == "" {
+			reqLogger.Info("Creating a new Secret", "Secret.Namespace", secret.Namespace, "Secret.Name", secret.Name)
+			err = r.client.Create(context.TODO(), secret)
+			if err != nil {
+				reqLogger.Error(err, "could not find or create identities Secret")
+				return reconcile.Result{}, err
+			}
```

Comment:
How about updating `infinispan.Spec.Security.EndpointSecretName` with the new name? So we inform the user on which secret is now in use

Reply:
I will check it, but looks like this will be updating in this code https://github.com/infinispan/infinispan-operator/blob/8c14d3e9631241648b6b785a4d011d5b164a5119/pkg/controller/infinispan/infinispan_controller.go#L352
during next reconcile

Reply:
that code applies only when the infinispan resources is updated. When the cluster is created with copy the Spec.Security into status.Security here:
https://github.com/infinispan/infinispan-operator/blob/8c14d3e9631241648b6b785a4d011d5b164a5119/pkg/controller/infinispan/infinispan_controller.go#L249

this is to prevent an immediate restart of the cluster.

Reply:
actually, there's probably something to check because I think I saw a restart

Reply:
Thanks, I will check